src/application.rs

```diff
@@ -139,9 +127,15 @@ pub fn run() {
                             Some(VirtualKeyCode::F) => {
                                 fullscreen_web();
                             }
+                            Some(VirtualKeyCode::Escape) if *state == ElementState::Pressed => {
+                                app.gui.change_state(Message::ShowControl(VisibleControl::ControlChoice));
+                            }
                             Some(VirtualKeyCode::Space) if *state == ElementState::Pressed => {
                                 experiment.toggle_pause();
                             }
+                            Some(VirtualKeyCode::D) if *state == ElementState::Pressed => {
+                                app.gui.change_state(Message::ToggleDebugMode);
+                            }
```

Comment:
Do you have to specify `if *state == ElementState::Pressed` every time? Doesn't the event only fire if a key gets pressed?

Reply:
Nope. I had toggles not working because the event came in for the un-pressed as well.

Reply:
Yeah I found the same.